src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -1218,7 +1243,7 @@ protected function isClassCastable($key)
     protected function isClassDeviable($key)
     {
         return $this->isClassCastable($key) &&
-            method_exists($castType = $this->parseCasterClass($this->getCasts()[$key]), 'increment') &&
+            method_exists($castType = $this->resolveCasterClass($key), 'increment') &&
```

Comment:
Is there a way to keep the behavior of not resolving the caster when using traditional caster classes? Then only resolve the custom caster resolver if a casterResolver has actually been registered? This code change makes it resolve a full instance every time.

Reply:
If you use DeviatesCastableAttributes, SerializesCastableAttributes for caster and use Castable - this does not work.

Reply:
Casters are  simple class usually. But I can add casterCache by field name in resolveCasterClass if need.

Reply:
@taylorotwell what you are think, maybe add caster cache on resolving a caster?